calico_upgrade/pkg/calicoupgrade.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/docopt/docopt-go"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/calico/calico_upgrade/pkg/commands"
+	"github.com/projectcalico/libcalico-go/lib/logutils"
+)
+
+func main() {
+	doc := `Usage:
+  calico-upgrade [options] <command> [<args>...]
+
+    dry-run   Perform a dry-run of the data migration. This validates that the
+              v1 formatted data will be successfully converted and that the v3
+              datastore is in the correct state for the data migration. This
+              command outputs a full report of any migrated names, migration
+              errors, or migrated name conflicts. See Description section
+              below for details.
```

Comment:
It seems that dry-run is also executed by "start". If so, what is the advantage of running it separately? That might be something interesting to mention IMO.